src/fetch_request.js

```diff
@@ -38,6 +38,18 @@ export class FetchRequest {
     this.options.headers = headers
   }
 
+  sameHostname () {
+    if (!this.originalUrl.startsWith('http:')) {
+      return true
+    }
+
+    try {
+      return new URL(this.originalUrl).hostname === window.location.hostname
+    } catch (_) {
+      return true
+    }
```

Comment:
This feels ugly, but `new URL()` bombs on relative paths and invalid URLs. Rather have soemthing further down the line fail because of a bad URL than it throwing an error here.